spyderlib/spyder.py

```diff
@@ -2517,15 +2535,21 @@ def apply_settings(self):
             default = default|QMainWindow.AnimatedDocks
         self.setDockOptions(default)
 
+        self.apply_panes_settings()
+        self.apply_statusbar_settings()
+
+    def apply_panes_settings(self):
+        """Update dockwidgets features settings"""
+        # Update toggle action on menu
         for child in self.widgetlist:
             features = child.FEATURES
             if CONF.get('main', 'vertical_dockwidget_titlebars'):
-                features = features|QDockWidget.DockWidgetVerticalTitleBar
+                features = features | QDockWidget.DockWidgetVerticalTitleBar
+            if not self.dockwidgets_locked:
+                features = features | QDockWidget.DockWidgetMovable
             child.dockwidget.setFeatures(features)
             child.update_margins()
 
-        self.apply_statusbar_settings()

```

Comment:
I refactor this part because to refresh the lock status of pane I could reuse some code, just scroll down and you will se I just move it down